src/Microsoft.AspNetCore.Antiforgery/Internal/DefaultClaimUidExtractor.cs

```diff
@@ -22,42 +23,94 @@ public DefaultClaimUidExtractor(ObjectPool<AntiforgerySerializationContext> pool
         }
 
         /// <inheritdoc />
-        public string ExtractClaimUid(ClaimsIdentity claimsIdentity)
+        public string ExtractClaimUid(ClaimsPrincipal claimsPrincipal)
         {
-            if (claimsIdentity == null || !claimsIdentity.IsAuthenticated)
+            if (claimsPrincipal == null)
+            {
+                throw new ArgumentNullException(nameof(claimsPrincipal));
+            }
```

Comment:
This is not a public API. Just `Debug.Assert()` when you want to check a parameter.